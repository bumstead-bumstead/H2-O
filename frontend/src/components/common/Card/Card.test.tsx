import Card from './Card';
import { render, screen } from '@/tests/test-util';

describe('Card 컴포넌트 테스트', () => {
  it('카드가 정상적으로 렌더링된다.', () => {
    render(<Card>카드</Card>);

    expect(screen.getByText('카드')).toBeInTheDocument();
  });

  describe('Card 스냅샷 테스트', () => {
    it('카드 스냅샷 테스트', () => {
      render(<Card>카드</Card>);

      expect(screen.getByText('카드')).toMatchSnapshot();
    });

    it('isSelected가 true일 때 스냅샷 테스트', () => {
      render(<Card isSelected>카드</Card>);

      expect(screen.getByText('카드')).toMatchSnapshot();
    });
  });
});
