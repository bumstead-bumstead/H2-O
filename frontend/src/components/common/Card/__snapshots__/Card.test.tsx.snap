// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card 컴포넌트 테스트 > Card 스냅샷 테스트 > isSelected가 true일 때 스냅샷 테스트 1`] = `
.emotion-0 {
  border: 1px solid #00AAD2;
  background-color: #00AAD21A;
  color: #212122;
  cursor: pointer;
}

.emotion-0:hover {
  border-color: #00AAD2;
}

<div
  class="emotion-0"
>
  카드
</div>
`;

exports[`Card 컴포넌트 테스트 > Card 스냅샷 테스트 > 카드 스냅샷 테스트 1`] = `
.emotion-0 {
  border: 1px solid #DADCDD;
  background-color: white;
  color: #A3A6AA;
  cursor: pointer;
}

.emotion-0:hover {
  border-color: #00AAD2;
}

<div
  class="emotion-0"
>
  카드
</div>
`;
